sympy/polys/domains/modularinteger.py

```diff
@@ -136,7 +136,7 @@ def __pow__(self, exp):
         else:
             val = self.val
 
-        return self.__class__(val**exp)
+        return self.__class__(pow(int(val), exp, self.mod))
```

Comment:
Is it not necessary to make the other args explicit integers?

Reply:
`pow` is dispatched on the first argument. The `int.__pow__` coerces the other two arguments automatically, so it isn't necessary. See https://docs.python.org/3/reference/datamodel.html#object.__pow__ and https://docs.python.org/3/library/functions.html#pow